src/jshint.js

```diff
@@ -1424,6 +1429,12 @@ var JSHINT = (function() {
       } else if ((!left.identifier || isReserved(left)) && left.id !== "." && left.id !== "[") {
         warning("W017", this);
       }
+      if (left && left.identifier) {
```

Comment:
This is great, but I think you want to look at what the `this` token is --- it's cool to reference properties of a const value, and even change them, but you can't  use a postfix increment/decrement (which I think is what you're trying to prevent here).

Also like above, add a comment explaining what the code is accomplishing, it helps a lot =)